apps/dot_ext/loggers.py

```diff
@@ -142,17 +163,43 @@ def set_session_values_from_auth_flow_uuid(request, auth_flow_uuid):
             pass
 
 
-def update_instance_auth_flow_trace_with_code(auth_uuid, code):
+def set_session_auth_flow_trace_value(request, key, value):
+    '''
+    Set auth flow key value in the session.
+    '''
+    if request.session:
+        request.session[key] = value
+
+
+def update_instance_auth_flow_trace_with_code(auth_dict, code):
     '''
-    Update AuthFlowUuid instance with code value.
+    Update AuthFlowUuid instance with code, crosswalk_type and share_demographic_scopes values.
 
     CALLED FROM:  apps.dot_ext.views.authorization.AuthorizationView.form_valid()
     '''
+    auth_uuid = auth_dict.get('auth_uuid', None)
+    auth_crosswalk_type = auth_dict.get('auth_crosswalk_type', None)
+    auth_share_demographic_scopes = auth_dict.get('auth_share_demographic_scopes', None)
+
     try:
-        if code and len(code.strip()) != 0:
```

Comment:
may be quote this try except block with  with transaction.atomic() to prevent race condition.

Reply:
@JFU-GIT Just updated that and for another code block. TY!